contrib/opensearch-project/opensearch-go.v4/opensearch.go

```diff
@@ -0,0 +1,222 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+// Package opensearch provides tracing functions for tracing the opensearch-project/opensearch-go/v4 package (https://github.com/opensearch-project/opensearch-go).
+package opensearch
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strconv"
+
+	"github.com/opensearch-project/opensearch-go/v4"
+	"github.com/opensearch-project/opensearch-go/v4/opensearchtransport"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/namingschema"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry"
+)
+
+const (
+	componentName      = "opensearch-project/opensearch-go/v4"
+	defaultServiceName = "opensearch.client"
+)
+
+var (
+	// bodyCutoff specifies the maximum number of bytes that will be stored as a tag
+	// value obtained from an HTTP request or response body.
+	bodyCutoff = 5 * 1024
+
+	_ opensearchtransport.Interface    = (*transport)(nil)
+	_ opensearchtransport.Discoverable = (*transport)(nil)
+	_ http.RoundTripper                = (*roundTripper)(nil)
+)
+
+func init() {
+	telemetry.LoadIntegration(componentName)
+	tracer.MarkIntegrationImported("github.com/opensearch-project/opensearch-go/v4")
+}
+
+// WrapClient wraps OpenSearch client with tracing.
+func WrapClient(c *opensearch.Client, opts ...Option) {
```

Comment:
```suggestion
func TraceClient(c *opensearch.Client, opts ...Option) {
```

in the context of this repo, wrapping usually means the function would return another `*opensearch.Client`, and this function modifies the received one.